Frontend/src/Components/registerPage/RegisterForm.tsx

```diff
@@ -0,0 +1,138 @@
+import { useState } from 'react'
+import RegisterBtn from '../registerPage/RegisterBtn'
+import { faEye, faEyeSlash } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+
+export default function RegisterForm() {
+  const [userData, setUserData] = useState({
+    name: '',
+    email: '',
+    password: '',
+    password2: '',
+    areTermsAccepted: false,
+  })
+
+  const [isPasswordVisible, setIsPasswordVisible] = useState<boolean>(false)
+  const [isPassword2Visible, setIsPassword2Visible] = useState<boolean>(false)
+
+  function handleChange(e: any): void {
+    const { name, value, type, checked } = e.target
+    type === 'checkbox'
+      ? setUserData({
+          ...userData,
+          [name]: checked,
+        })
+      : setUserData({
+          ...userData,
+          [name]: value,
+        })
```

Comment:
Do logiki nie będę się niżej rozpisywał bo na pewno formy inaczej obsłużymy, ale zawsze warto pisać kod w miarę czytelnie np.

```suggestion
  setUserData({
          ...userData,
          [name]: type === 'checkbox' ? checked : value,
        })
```